File: backend/app/schema.py
from pydantic import BaseModel

from .tables import CheckinStatus


class CheckinCreate(BaseModel):
    status: CheckinStatus
    first_name: str
    last_name: str
    reservation_code: str


class Checkin(BaseModel):
    id: int
    status: CheckinStatus
    first_name: str
    last_name: str
    reservation_code: str

File: backend/src/schema.rs
table! {
    use diesel::sql_types::*;
    use crate::models::{CheckinStatus as Checkinstatus};

    checkins (id) {
        id -> Int4,
        first_name -> Varchar,
        last_name -> Varchar,
        reservation_code -> Varchar,
        status -> Checkinstatus,
        logs -> Nullable<Text>,
    }
}
